airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -196,3 +198,137 @@ def _is_global_state(state: Union[List[AirbyteStateMessage], MutableMapping[str,
     @staticmethod
     def _is_per_stream_state(state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]]) -> bool:
         return isinstance(state, List)
+
+
+class ConcurrencyCompatibleStateType(Enum):
+    date_range = "date-range"
+
+
+class ConcurrentConnectorStateManager(ConnectorStateManager, ABC):
+    START_KEY = "start"
```

Comment:
do we need a way to make this configurable, or to translate the keys between the in-memory state and the persisted state? eg if the existing state uses 'start_time' and 'end_time' instead of 'start' and 'end'?

Reply:
My assumption was that all already-concurrent states would have the same keys, so the translation was only necessary between legacy and concurrent - the translation for that happens in `convert_from_sequential_state`.

Reply:
oh I see. I suppose the "created" field should be configurable in the `EpochValueConcurrentStreamStateConverter` since that field in source-specific

Reply:
@girarda are the sources' state messages consistent enough that we can assume they'll have an equivalent field?

Reply:
yes. only considering non-partitioned streams, the state is typically represented as
```
{
"{stream_name}": {
     "{cursor_field}": "{formatted_timestamp}"
  }
}
```

Couple of examples:
- [amazon ads](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py#L365)
- [stripe](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-stripe/source_stripe/streams.py#L223)
- [salesforce](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L701)

Reply:
Sweet, thanks for the examples!

Reply:
Updated to make the cursor field configurable.